Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -488,7 +488,25 @@ namespace Babylon
             m_frame = m_session->GetNextFrame(shouldEndSession, shouldRestartSession);
             m_frame.reset();
         } while (!shouldEndSession);
-        // Clear frameBufferData and destroy bgfx FrameBuffers
+
+        // Destroy any bgfx FrameBuffers that are not marked as owned by JS.
+        bool frameBuffersDestroyed{false};
+        for (auto& pair : m_texturesToFrameBuffers)
+        {
+            if (!pair.second->OwnedByJS)
+            {
+                frameBuffersDestroyed = true;
+                delete pair.second;
+            }
+        }
+
+        // If any frame buffers were destroyed, we need to call bgfx::frame once to force clean up.
+        // This is likely broken for multi-threaded scenarios.
+        if (frameBuffersDestroyed)
```

Comment:
This should probably warrant a new issue or at least a TODO.

Reply:
Created issue here: https://github.com/BabylonJS/BabylonNative/issues/555